cpp/src/plasma/CMakeLists.txt

```diff
@@ -123,6 +123,18 @@ endif()
 
 add_executable(plasma_store store.cc)
 target_link_libraries(plasma_store plasma_static ${PLASMA_LINK_LIBS})
+SET_TARGET_PROPERTIES(plasma_store PROPERTIES
+  LINK_FLAGS "-Wl,--as-needed")
+
+if (ARROW_RPATH_ORIGIN)
```

Comment:
I think this block could benefit from a short comment explaining what the purpose is.

Reply:
This looks like a rebase artifact

Reply:
Yes, I have added this in another commit (without it, the manylinux1 wheel would fail).